plugins/wasi_nn/wasinn_whisper.cpp

```diff
@@ -16,6 +18,309 @@ namespace WasmEdge::Host::WASINN::Whisper {
 #ifdef WASMEDGE_PLUGIN_WASI_NN_BACKEND_WHISPER
 
 namespace {
+int timestampToSample(int64_t T, int NSamples, int WhisperSampleRate) {
+  return std::max(0, std::min(static_cast<int>(NSamples) - 1,
+                              static_cast<int>((T * WhisperSampleRate) / 100)));
+}
+std::string toTimestamp(int64_t T, bool Comma) {
+  int64_t Msec = T * 10;
+  int64_t Hr = Msec / (1000 * 60 * 60);
+  Msec = Msec - Hr * (1000 * 60 * 60);
+  int64_t Min = Msec / (1000 * 60);
+  Msec = Msec - Min * (1000 * 60);
+  int64_t Sec = Msec / 1000;
+  Msec = Msec - Sec * 1000;
+
+  char Buf[32];
+  snprintf(Buf, sizeof(Buf), "%02d:%02d:%02d%s%03d", static_cast<int>(Hr),
+           static_cast<int>(Min), static_cast<int>(Sec), Comma ? "," : ".",
+           static_cast<int>(Msec));
+
+  return std::string(Buf);
+}
+
+std::string
+estimateDiarizationSpeaker(const std::vector<std::vector<float>> PCMF32s,
+                           int64_t T0, int64_t T1, bool IdOnly = false) {
+  std::string Speaker = "";
+  const int64_t NSamples = PCMF32s[0].size();
+
+  const int64_t is0 = timestampToSample(T0, NSamples, WHISPER_SAMPLE_RATE);
+  const int64_t is1 = timestampToSample(T1, NSamples, WHISPER_SAMPLE_RATE);
```

Comment:
`Is0`, `Is1`.
Could you explain the logic about the `is` and `energy?

Reply:
I implemented it based on whisper's example. This feature is for stereo audio. I think `is` means the start and end of the sound segment. Then, calculate the PCMF signal stored to `energy` to know where the sound is from.